src/audio/buffer.c

```diff
@@ -151,12 +164,14 @@ void comp_update_buffer_produce(struct comp_buffer *buffer, uint32_t bytes)
 	}
 
 	irq_local_disable(flags);
+	buffer_lock(buffer);
 
 	audio_stream_produce(&buffer->stream, bytes);
 
 	notifier_event(buffer, NOTIFIER_ID_BUFFER_PRODUCE,
 		       NOTIFIER_TARGET_CORE_LOCAL, &cb_data, sizeof(cb_data));
```

Comment:
I hope there is no issue with only notifying locally.

Reply:
@paulstelian97 Probably it will need to be changed to multicore notification.

Reply:
Can you remind me, who is actually listening on these produce/consume events?

Reply:
Currently there are no clients for those calls, that's why I said "probably". But since there are no clients and no actual components running on slave cores I would say it's a low priority for now.

Reply:
Alright, still this needs to be kept in mind for the future (it's a way to get weird bugs in the future, we may forget about this)